lib/lab_manager/app/endpoints/computes.rb

```diff
@@ -132,11 +132,22 @@ class Compute < Base
 
         get '/:id/snapshots/:snapshot_id' do
           compute = ::Compute.find(params[:id])
-          action = compute.snapshots.find(params[:snapshot_id])
-          action.to_json
+          snapshot = compute.snapshots.find(params[:snapshot_id])
+          snapshot.to_json
         end
 
         get '/:id/snapshots/:snapshot_id/revert' do
+          compute = ::Compute.find(params[:id])
+          snapshot = compute.snapshots.find(params[:snapshot_id])
+
+          action = compute.actions.create!(
+            command: 'revert_snapshot_vm',
+            payload: {
+              snapshot_id: snapshot.id,
+            }
+          )
```

Comment:
For a nice design wouldn't we encapsulate this logic somewhere? We are reverting snapshot, but actually we are creating an action on compute. It's breaking several design patters. (The same counts for create snapshot.)
We can discuss personally about it on Monday.